hopp/simulation/technologies/wind/floris.py

```diff
@@ -239,27 +265,90 @@ def export(self):
-        self.fi.set(layout_x=xcoords, layout_y=ycoords)
+        self.fi.set(
+            layout_x = xcoords,
+            layout_y = ycoords
+            )
         self.nTurbs = len(xcoords)
-        self.system_capacity = len(xcoords) * self.turb_rating
+        self.system_capacity = len(xcoords)*self.turb_rating
         self.value("wind_farm_xCoordinates", xcoords)
         self.value("wind_farm_yCoordinates", ycoords)     
+    
+    def initialize_wind_turbine(self, floris_config): 
+        """Update `floris_config` with turbine parameters. Checks the turbine library 
+            and floris internal library for a turbine with name matching either 
+            `config.turbine_name` or `floris_config["farm"]["turbine_type"][0]["turbine_type"]`.
+            If no matching turbine is found, find the turbine with the closest 
+            matching name and raise a warning.
+
+        Args:
+            floris_config (dict): floris (version 4) input dictionary.
+
+        Raises:
+            UserWarning: if turbine name does not exactly match a turbine in 
+                the turbine library or floris library.
+
+        Returns:
+            dict: floris config with turbine model parameters updated in `floris_config["farm"]["turbine_type"][0]`
+        """
+        if self.config.turbine_name is None:
+            
+            # turbine data is included in floris_config
+            if isinstance(floris_config["farm"]["turbine_type"][0],dict):
+                self.turbine_name = floris_config["farm"]["turbine_type"][0]["turbine_type"]
+
+            # load file from internal floris library
+            if isinstance(floris_config["farm"]["turbine_type"][0],str):
+                turbine_lib_res = floris_tools.check_libraries_for_turbine_name_floris(floris_config["farm"]["turbine_type"][0], self)
+                if isinstance(turbine_lib_res,str):
+                    raise UserWarning(turbine_lib_res)
+                floris_config["farm"]["turbine_type"][0] = turbine_lib_res
+                
+        else:
+            turbine_lib_res = floris_tools.check_libraries_for_turbine_name_floris(self.config.turbine_name, self)
+            if isinstance(turbine_lib_res,str):
+                turbine_name = check_turbine_name(self.config.turbine_name)
+                turbine_lib_res = get_floris_turbine_specs(turbine_name,self)
+                logger.warning(f"closest matching turbine name to {self.config.turbine_name} is {turbine_name} ... setting turbine model as {turbine_name}")
+            floris_config["farm"]["turbine_type"][0] = turbine_lib_res 
+        return floris_config
+
+    def update_wind_turbine(self,turbine_name):
+        """Update `FlorisModel` (`self.fi`) with with turbine parameters corresponding
+            to `turbine_name`. Used to update turbine parameters after Floris has been initialized.
+            Updates system capacity, rotor diameter, power-curve, and turb_rating.
```

Comment:
Can you remove one indent on these two lines? They just need to be in line with the first quote.